Resources/Storage/StorageElement.py

```diff
@@ -397,39 +395,72 @@ def getOccupancy(self, unit='MB', **kwargs):
     filteredPlugins = self.__filterPlugins('getOccupancy')
     if not filteredPlugins:
       return S_ERROR(errno.EPROTONOSUPPORT, "No storage plugins to query the occupancy")
+
+    # Call occupancy plugin if requested
+    occupancyPlugin = self.options.get('OccupancyPlugin')
+    if occupancyPlugin:
+      res = ObjectLoader().loadObject(occupancyPlugin)
+      if not res['OK']:
+        return S_ERROR(errno.EPROTONOSUPPORT, 'Failed to load occupancy plugin %s' % occupancyPlugin)
+      log.verbose('Use occupancy plugin %s' % occupancyPlugin)
+      occupancyPlugin = res['Value'](self)
+      if occupancyPlugin.isUsable:
+        res = occupancyPlugin.getOccupancy()
+        if res['OK']:
+          occupancyDict = res['Value']
+          result = self.checkOccupancy(occupancyDict, unit)
+          if result['OK']:
+            return result['Value']
+
     # Try all of the storages one by one
     for storage in filteredPlugins:
 
       # The result of the plugin is always in B
       res = storage.getOccupancy(**kwargs)
       if res['OK']:
         occupancyDict = res['Value']
-
-        # Make sure all the mandatory parameters are present
-        if set(occupancyDict) & mandatoryParams != mandatoryParams:
-          log.verbose("Missing mandatory parameters", mandatoryParams - set(occupancyDict))
+        result = self.checkOccupancy(occupancyDict, unit)
+        if not result['OK']:
           continue
-
-        # Since plugins return Bytes, we do not need to convert if that's what we want
-        if unit != 'B':
-          for space in ['Total', 'Free']:
-            convertedSpace = convertSizeUnits(occupancyDict[space], 'B', unit)
-            # If we have a conversion error, we go to the next plugin
-            if convertedSpace == -sys.maxsize:
-              log.verbose(
-                  "Error converting %s space from MB to %s: %s" %
-                  (space, unit, occupancyDict[space]))
-              break
-            occupancyDict[space] = convertedSpace
-
-        # If the plugin does not return the SpaceReservation, take it from the CS
+        occupancyDict = result['Value']
 
         if 'SpaceReservation' not in occupancyDict:
           occupancyDict['SpaceReservation'] = self.options.get('SpaceReservation')
         return res
```

Comment:
This `return res` only works because the `occupancyDict` is modified in `checkOccupancy`, if `checkOccupancy` were not to modify the `occupancyDict`, this would not return the updated dictionary. Maybe it would be more clear to just `return S_OK(occupancyDict)`

Reply:
I hesitated to change it, since this behavior was inherited from the original code which updates `occupancyDict` while returns `res` given by `res = storage.getOccupancy(**kwargs)`. 
If no objection, I'd follow your suggestion.